tests.ml

```diff
@@ -151,10 +160,40 @@ let test_broken_scope_3 =
     print x;
   |]
 
```

Comment:
@gasche oh I see, I did not realize that the assembler also understands the annotations now. But the parser/lexer is essentially untested, right?
but lets merge this since the parser/lexer is already in anyways.